examples/workflows/chunk_clustering.py

```diff
@@ -0,0 +1,240 @@
+        self, 
+        field: str,
+        output_chunk_field: str="sentence_chunk_",
+        language: str='en'):
+        self.split_function = partial(split_text_into_sentences, language=language)
+        self._field = field
+        self._output_chunk_field = output_chunk_field
+        super().__init__(
+            input_fields=[self._field],
+            output_fields=[self._output_chunk_field]
+        )
+
+        
+    def transform(self, documents: DocumentList) -> DocumentList:
+        """
+        Split sentences into output chunk fields
+        """
+        [
+            d.split(
+                self.split_function, 
+                chunk_field=self._output_chunk_field,
+                field=self._field
+            ) for d in documents
+        ]
+        return documents
+
+class ChunkVectorizerOperator(AbstractOperator):
+    def __init__(self, text_field: str, chunk_field: str,
+        output_field: str):
+        self._text_field = text_field
+        self._chunk_field = chunk_field
+        self._output_field = output_field
+        super().__init__(
+            input_fields=[self._chunk_field],
+            output_fields=[self._chunk_field]
+        )
+    
+    def _random_vector(self):
+        return [random.randint(0, 99) for _ in range(10)]
+    
+    def _vectorize(self, chunk_values, *args, **kwargs):
+        return [self._random_vector() for _ in range(len(chunk_values))]
+        
+    def transform(self, documents: DocumentList) -> DocumentList:
+        """
+        Split sentences
+        """
+        [
+            d.operate_on_chunk(
+                operator_function=self._vectorize,
+                chunk_field=self._chunk_field,
+                field=self._text_field,
+                output_field=self._output_field
+            )
+            for d in documents
+        ]
+        return documents
+
+class ChunkClusterOperator(AbstractOperator):
+    def __init__(
+        self,
+        n_clusters: int,
+        chunk_field: str,
+        vector_field: str,
+        alias: Optional[str] = None,
+    ):
+        self._model = KMeans(n_clusters=n_clusters)
+
+        self._chunk_field = chunk_field
+        self._vector_field = vector_field
+        self._alias = f"kmeans-{n_clusters}" if alias is None else alias
+        self._output_field = f"_cluster_.{vector_field}.{self._alias}"
+
+        super().__init__(
+            input_fields=[self._vector_field],
+            output_fields=[self._output_field],
+        )
+
+    def transform(self, documents: DocumentList) -> DocumentList:
+        """
+        Main transform function
+        """
+        vectors = documents.get_chunks_as_flat(
+            chunk_field=self._chunk_field, field=self._vector_field
+        )
+        vectors = np.array(vectors)
+        labels = self._model.fit_predict(vectors).tolist()
+
+        for document, label in zip(documents, labels):
+            document[self._output_field] = f"cluster_{label}"
```

Comment:
The issue here is that we aren't storing this as an array properly. We need to make sure that the output field in this case is an actual array